src/cli/commands/global.js

```diff
@@ -159,18 +160,38 @@ async function initUpdateBins(config: Config, reporter: Reporter, flags: Object)
       }
     }
 
-    // add new bins
-    for (const src of afterBins) {
```

Comment:
Wouldn't it be fixed simply but filtering `afterBins` here to make sure we don't iterate over `.ps1` and `.cmd` files?

Reply:
It would still be creating shims of shims. It would fix the `.ps1.cmd`/`.cmd.ps1`/etc issue, but it would not fix the problem of there being shims pointing to shims pointing to binaries. `cmd-shim` does not work correctly when such a condition occurs, at least not with PowerShell.

Reply:
Longer explanation:

The `cmd-shim` package looks at the shebang of the program it's targeting to figure out what to write into the shim. This works fine for those two, because for bash it will always be either `node` or `sh` and for cmd it will always be either `node.exe` or `cmd.exe`. PowerShell breaks because the shebang on the PowerShell shim ends up referencing `pwsh.exe`, which is the binary for PowerShell Core, even when PowerShell Core is not installed. When a shim is created to a PowerShell shim, it automatically references `pwsh` even when PowerShell Core isn't available on the system.

I suppose it might be worth it to file an issue with `cmd-shim` to write the *correct* shebang to PowerShell shims, but I also would have to argue that it's still best to have one layer of shims if possible instead of shims-pointing-to-shims.